Source/WebKit/UIProcess/WebFramePolicyListenerProxy.h

```diff
@@ -71,6 +76,8 @@ class WebFramePolicyListenerProxy : public API::ObjectImpl<API::Object::Type::Fr
     std::optional<RefPtr<BrowsingWarning>> m_safeBrowsingWarning;
     std::optional<std::optional<NavigatingToAppBoundDomain>> m_isNavigatingToAppBoundDomain;
     bool m_doneWaitingForLinkDecorationFilteringData { false };
+    bool m_doneWaitingForSafeBrowsing { false };
+    RunLoop::Timer m_safeBrowsingTimeout;
```

Comment:
This seems like it's not really the responsibility of the listener, but more the responsibility of the WebPageProxy.